### PR TITLE
Detect library changes at startup

### PR DESCRIPTION
**Pull request by MatteO-Matic**
Re-scan the local music folder at startup for any changes made while the application wasn't running.
related to https://bugs.launchpad.net/noise/+bug/1073871

### CODE REVIEW

**Comment by matzipan** [comment]
I have pushed some minor commits and added some braces. Please always add braces - [even if its only one line](https://elementary.io/fr/docs/code/reference#indentation).

File: src/LocalBackend/LocalLibrary.vala [hunk 1]
```diff
@@ -257,36 +257,42 @@ public class Noise.LocalLibrary : Library {
...
+        foreach (var m in get_medias()) {
+            if (!m.isTemporary && !m.isPreview && m.uri.contains (music_folder_dir)) {
+                if (!File.new_for_uri (m.uri).query_exists ()) {
+                    to_remove.add (m);
+                }
+
+                if (files.contains (m.uri)) {
+                    files.remove (m.uri);
+                }
+            }
```
Comment (matzipan) [inline src/LocalBackend/LocalLibrary.vala]: AFAICT that is not what the original code did. The original code was:
```
            if (!m.isTemporary && !m.isPreview && m.uri.contains (music_folder_dir))

            if (!File.new_for_uri (m.uri).query_exists ())
                to_remove.add (m);
            if (files.contains (m.uri))
                files.remove (m.uri);
```

Which is the equivalent of:
```
            if (!m.isTemporary && !m.isPreview && m.uri.contains (music_folder_dir)) {
                if (!File.new_for_uri (m.uri).query_exists ()) {
                    to_remove.add (m);
                }
            }
            if (files.contains (m.uri)) {
                files.remove (m.uri);
             }
```

Is there any reason for changing this?

**Comment by matzipan** [comment]
@MatteO-Matic  don't forget to claim the bounty here  https://www.bountysource.com/issues/280973-monitor-library-for-changes

**Comment by MatteO-Matic** [comment]
@matzipan Someone would need to close the issue on launchpad for me to collect the bounty.

**Comment by danrabbit** [comment]
Closed. It might take a day for Bountysource to update

**Pull request merged**